includes/functions/functions_osh_update.php

```diff
@@ -0,0 +1,191 @@
+                    OSH_EMAIL_TEXT_STATUS_PLEASE_REPLY;
+           
+                $html_msg['EMAIL_CUSTOMERS_NAME']    = $osh_info->fields['customers_name'];
+                $html_msg['EMAIL_TEXT_ORDER_NUMBER'] = OSH_EMAIL_TEXT_ORDER_NUMBER . ' ' . $orders_id;
+                $html_msg['EMAIL_TEXT_INVOICE_URL']  = '<a href="' . zen_catalog_href_link(FILENAME_CATALOG_ACCOUNT_HISTORY_INFO, "order_id=$orders_id", 'SSL') .'">' . str_replace(':', '', OSH_EMAIL_TEXT_INVOICE_URL) . '</a>';
+                $html_msg['EMAIL_TEXT_DATE_ORDERED'] = OSH_EMAIL_TEXT_DATE_ORDERED . ' ' . zen_date_long($osh_info->fields['date_purchased']);
+                $html_msg['EMAIL_TEXT_STATUS_COMMENTS'] = nl2br($email_message);
+                $html_msg['EMAIL_TEXT_STATUS_UPDATED'] = str_replace("\n", '', $status_text);
+                $html_msg['EMAIL_TEXT_STATUS_LABEL'] = str_replace("\n", '', $status_value_text);
+                $html_msg['EMAIL_TEXT_NEW_STATUS'] = $new_orders_status_name;
+                $html_msg['EMAIL_TEXT_STATUS_PLEASE_REPLY'] = str_replace("\n", '', OSH_EMAIL_TEXT_STATUS_PLEASE_REPLY);
+                $html_msg['EMAIL_PAYPAL_TRANSID'] = '';
+                
+                if (empty($email_subject)) {
+                    $email_subject = OSH_EMAIL_TEXT_SUBJECT . ' #' . $orders_id;
+                }
+
+                if ($notify_customer == 1) { 
+                    zen_mail($osh_info->fields['customers_name'], $osh_info->fields['customers_email_address'], $email_subject, $email_text, STORE_NAME, EMAIL_FROM, $html_msg, 'order_status');
+                } 
+
+                // PayPal Trans ID, if any
+                $result = $GLOBALS['db']->Execute(
+                    "SELECT txn_id, parent_txn_id 
+                       FROM " . TABLE_PAYPAL . " 
+                      WHERE order_id = $orders_id
+                   ORDER BY last_modified DESC, date_added DESC, parent_txn_id DESC, paypal_ipn_id DESC"
+                );
+                if (!$result->EOF) {
+                    $email_text .= "\n\n" . ' PayPal Trans ID: ' . $result->fields['txn_id'];
+                    $html_msg['EMAIL_PAYPAL_TRANSID'] = $result->fields['txn_id'];
+                }
+
+                //send extra emails
+                if (empty($send_xtra_emails_to) && SEND_EXTRA_ORDERS_STATUS_ADMIN_EMAILS_TO_STATUS == '1') {
+                    $send_xtra_emails_to = (string)SEND_EXTRA_ORDERS_STATUS_ADMIN_EMAILS_TO;
+                }
+                if (!empty($send_xtra_emails_to)) {
+                    zen_mail('', $send_xtra_emails_to, SEND_EXTRA_ORDERS_STATUS_ADMIN_EMAILS_TO_SUBJECT . ' ' . $email_subject, $email_text, STORE_NAME, EMAIL_FROM, $html_msg, 'order_status_extra');
+                }
+            }
+    
+            if (empty($updated_by)) {
+                if (IS_ADMIN_FLAG === true && isset($_SESSION['admin_id'])) {
+                    $updated_by = zen_updated_by_admin();
```

Comment:
@lat9 this `zen_updated_by_admin()` function appears to be missing.

Reply:
Darn it; let me get that included and review that other function dependencies.